### PR TITLE
feat: 🎸 incremental

### PR DESCRIPTION
**Pull request by IWANABETHATGUY**

### CODE REVIEW

**Comment by IWANABETHATGUY** [comment]
related to https://github.com/johnsoncodehk/volar/projects/1#card-50201488

**Comment by IWANABETHATGUY** [comment]
image
now, every didChange onlySend documentChange instead of whole document

**Comment by IWANABETHATGUY** [comment]
add a configuration for more convenient debugging
https://code.visualstudio.com/api/language-extensions/language-server-extension-guide#logging-support-for-language-server

File: packages/client/src/extension.ts [hunk 1]
```diff
@@ -89,6 +89,7 @@ function createLanguageService(context: vscode.ExtensionContext, mode: string, n
 
 	// Create the language client and start the client.
 	const client = new lsp.LanguageClient(
+		id,
```
Comment (IWANABETHATGUY) [inline packages/client/src/extension.ts:92]: add language id , which is required for trace configuration

**Comment by johnsoncodehk** [comment]
thanks for the improve! I need some time to check the behavior for `TextDocumentSyncKind.Incremental`.

btw I'm refactoring the whole project on refactor branch, all code will change, so don’t make too deep changes for now (this pr is OK!).

**Comment by johnsoncodehk** [comment]
Look good, thanks!

**Pull request merged**